tools/debug/debugsymbolviewer.py

```diff
@@ -0,0 +1,178 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+
+from __future__ import print_function
+import re
+
+systemmap_path = '../../build/output/bin/System.map'	# System map file
+log_file = './log_file'		# Crash log file
+symbol_lookup_table = []	# Lookup table created from contents of system map file
+stack_table = []		# Lookup table created from stack contents of assert log file
+stacksize = 0			# Stack size set to 0 and it gets updated dynamically while using
+stack_start_addr = 0x000000	# Default stack start address
+debug = False			# Variable to print debug logs
+pc_value = 0x000000		# Program counter value
+
+# Kernel text start & end addresses
+ram_text_start_addr = 0x000000
+ram_text_end_addr = 0x000000
+flash_text_start_addr = 0x000000
+flash_text_end_addr = 0x000000
+
+# Function to setup the Address to Symbol mapping table from the System Map file
+def setup_symbol_table(sm_file):
+	global flash_text_start_addr
+	global flash_text_end_addr
+	global ram_text_start_addr
+	global ram_text_end_addr
+	# Reading the System Map file and preparing the symbol map table
+	with open(sm_file) as searchfile:
+		for line in searchfile:
+			s = line.split(' ', 2)
+			if len(s) == 3:
+				if not '$' in s[2]:
+					# s[0] contains address and s[2] contains Symbol name
+					symbol_lookup_table.append((int(s[0], 16), s[2].rstrip()))
+
+					# Extract kernel text start & end addresses from the table
+					symbol = s[2].rstrip('\n')
+					if symbol == '_stext':
+						flash_text_start_addr = hex(int(s[0], 16))
+						#flash_text_start_addr = s[0]
+					if symbol == '_etext':
+						flash_text_end_addr= hex(int(s[0], 16))
+						#flash_text_end_addr = s[0]
+					if symbol == '_ram_kernel_text_start':
+						ram_text_start_addr = hex(int(s[0], 16))
+						#ram_text_start_addr = s[0]
+					if symbol == '_ram_kernel_text_end':
+						ram_text_end_addr = hex(int(s[0], 16))
+						#ram_text_end_addr = s[0]
+
+	if debug:
+		# Printing the Created Symbol table
+		print('~~~~~~~~~~~~~~~~~~~~~~~~ SYMBOL TABLE START ~~~~~~~~~~~~~~~~~~~~~')
+		for line in symbol_lookup_table:
+			print("{0:x} {1}".format(line[0], line[1]))
+		print('~~~~~~~~~~~~~~~~~~~~~~~~ SYMBOL TABLE END   ~~~~~~~~~~~~~~~~~~~~~')
+
+		# Printing the kernel text address ranges in ram and flash areas
+		print('~~~~~~~~~~~~~~~~~~~~~~~~~~  KERNEL TEXT RANGE ~~~~~~~~~~~~~~~~~~~~~~~')
+		print('flash_text_start_addr\t=\t', flash_text_start_addr)
+		print('flash_text_end_addr\t=\t', flash_text_end_addr)
+		print('ram_text_start_addr\t=\t', ram_text_start_addr)
```

Comment:
Only some boards have ram text region. Is there any way to check and print only it has?